kernel/main.c

```diff
@@ -340,6 +353,52 @@ async_init(void)
       panic("Unable to create a kthread for do_async_init()");
 }
 
+static void
+setup_uefi_runtime_services() {
+   EFI_RUNTIME_SERVICES *RT = TO_PTR(uefi_rt_addr);
+
+   printk("Runtime Services: %p\n", RT);
+
+   struct mem_region ma;
```

Comment:
Cosmetic: declare the stack variables in the main scope at the beginning on the function. That allows with a quick look to get an idea of how much stack space a function is used. Declaring variables sparsely in the body is more "polluted" and doesn't help in C because we don't have RAII as in C++. Writing C code might look like a more "low-level" version of C++, but in reality is a whole different world. Different mindset etc. 

Sadly, often C++ programmers don't understand C programmers and vice-versa and end up almost "hating" each other because they believe the others are doing "stupid" things. While in reality, the "others" are living in a completely different world.

Reply:
I'm slowly beginning to understand this difference in mindsets, it's fascinating